zipline/utils/calendars/trading_schedule.py

```diff
@@ -0,0 +1,379 @@
+        return self.schedule.index
+
+    @property
+    @remember_last
+    def all_execution_minutes(self):
+        return all_scheduled_minutes(self.all_execution_days,
+                                     self.execution_minutes_for_days_in_range)
+
+    def trading_dates(self, start, end):
+        """
+        Calculates the dates of all of the trading sessions between the given
+        start and end.
+
+        Parameters
+        ----------
+        start : Timestamp
+        end : Timestamp
+
+        Returns
+        -------
+        DatetimeIndex
+            A DatetimeIndex containing the dates of the desired trading
+            sessions.
+        """
+        return self.trading_sessions(start, end).index
+
+    @abstractmethod
+    def data_availability_time(self, date):
+        """
+        Given a UTC-canonicalized date, returns a time by-which all data from
+        the previous date is available to the algorithm.
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized calendar date whose data availability time
+            is needed.
+
+        Returns
+        -------
+        Timestamp or None
+            The data availability time on the given date, or None if there is
+            no data availability time for that date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def start_and_end(self, date):
+        """
+        Given a UTC-canonicalized date, returns a tuple of timestamps of the
+        start and end of the algorithm trading session for that date.
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized algorithm trading session date whose start
+            and end are needed.
+
+        Returns
+        -------
+        (Timestamp, Timestamp)
+            The start and end for the given date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def is_executing_on_minute(self, dt):
+        """
+        Calculates if a TradingAlgorithm using this TradingSchedule should be
+        executed at time dt.
+
+        Parameters
+        ----------
+        dt : Timestamp
+            The time being queried.
+
+        Returns
+        -------
+        bool
+            True if the TradingAlgorithm should be executed at dt,
+            otherwise False.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def is_executing_on_day(self, dt):
+        """
+        Calculates if a TradingAlgorithm using this TradingSchedule would
+        execute on the day of dt.
+
+        Parameters
+        ----------
+        dt : Timestamp
+            The time being queried.
+
+        Returns
+        -------
+        bool
+            True if the TradingAlgorithm should be executed at dt,
+            otherwise False.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def minute_window(self, start, count, step=1):
```

Comment:
Do we not have enough information to implement this in terms of the abstract intefaces of the class?  (It seems like we should be able to do so)